### ISSUE TITLE
"Keep app open" card

### ISSUE THREAD

**Issue by JeanlChristophe** [synonymdev/bitkit-react-native#2190]
The toast is still there even atfer the channel has been created.
Toast still display after a force close app.

![Image](https://github.com/user-attachments/assets/8490d232-4b40-4826-a6f4-bbadf3922be9)

**Comment by pwltr** [comment]
This card is for when a channel couldn't be closed after a transfer to savings. It will keep trying for 30m and then offer to force close. Was your channel closed @JeanlChristophe ?

**Comment by JeanlChristophe** [comment]
It's been more than 48hours. No channels closed at the moment. I only have 1 channel close but dated from the 23rd of august.
I have 10 channels open at the moment, impossible to know which channels I asked to close. All the status are "Connection open".
![Screenshot_20240902-154633](https://github.com/user-attachments/assets/83d80467-2c65-4d02-98a4-f3ad4c30e7f0)
![Screenshot_20240902-154641](https://github.com/user-attachments/assets/8fb39831-ae32-463d-b561-533812eb63e8)
![Screenshot_20240902-154656](https://github.com/user-attachments/assets/4f6c6130-74e6-4dd8-a091-5324efa085ef)



Please find my logs if needed
[bitkit_ldk_logs_1725284859455 (1).zip](https://github.com/user-attachments/files/16838350/bitkit_ldk_logs_1725284859455.1.zip)

**Comment by pwltr** [comment]
It kinda looks like your channels are not working correctly. According to logs you have 8 channels of which 4 are not usable. That could explain why you can't close them. @Jasonvdb FYI

```
2024-08-31 20:06:13 ********NODE STATE********
NodeID: 0257daa02f027f835a8356afb7c5e2ee98731d2f97d34042a3e519a9d8050b0479
Open channel:
Funding txid: d9e9878173a60302cd40a731c57690abbcdb0d1a63aef5bcd5309ebc45a7ccc4
Ready: YES
Usable: NO
Balance: 40979 sats
Open channel:
Funding txid: caa882260011715df021955752f26558c6b59969257da3ab7bf0e3da98095a43
Ready: YES
Usable: NO
Balance: 0 sats
Open channel:
Funding txid: 57cff1f9651d0cbd217183e6b8fc649fc0987da3757c284c9b70294d4e532305
Ready: YES
Usable: NO
Balance: 0 sats
Open channel:
Funding txid: 92fe66041c3e2ef93f699ba9f4d126fabdf07a9188ab499ec38a4ddd60255ad3
Ready: YES
Usable: NO
Balance: 0 sats
Open channel:
Funding txid: dd0672dd8b8aeb54559531955d383900a8a232138d1b6fe2b8ce6872d26a0f97
Ready: YES
Usable: YES
Balance: 28579 sats
Open channel:
Funding txid: e67540d0b09f32f7738b7239b7fa0f5f8d31a5310b3fe97cdce820c366d864b0
Ready: YES
Usable: YES
Balance: 13063 sats
Open channel:
Funding txid: 25a303927b408402a32da53f621147d8354691c420b795ba62b1208bcf2eac70
Ready: YES
Usable: YES
Balance: 20501 sats
Open channel:
Funding txid: 4959b9e91c05ac3b2eec40b5e484b78224bf2357b5633bf8098c66583fe2e304
Ready: YES
Usable: YES
Balance: 36512 sats
```

**Comment by Jasonvdb** [comment]
I'm guessing 4 of these channels may have been from an external LSP? Usually if they are not usable then the peer is not connected. If you can remember where you got these channels you can add the peer again and then try a mutual close. Else it will have to end in a force close

**Comment by Jasonvdb** [comment]
Maybe you did a restore and peer was not restored. Related to this issue https://github.com/orgs/synonymdev/projects/12/views/1?pane=issue&itemId=69388158

**Comment by JeanlChristophe** [comment]
Ok, let's doing it this sprint otherwize it can be a nightmare for people that are using other LSP.

**Comment by JeanlChristophe** [comment]
https://github.com/user-attachments/assets/7bb21f80-39bd-49e6-9f6a-658cc59355df

Channels are closed now but I still have this card @pwltr

**Comment by JeanlChristophe** [comment]
Ps: I don't have anymore this card once I restore the wallet.

**Comment by catch-21** [comment]
I suspect this either happens when closing any manually setup connect or when the 'Funds in Transfer' screen is dismissed before it progresses to 'Funds moved to savings'. To be confirmed.

**Comment by catch-21** [comment]
Tested on 1.0.5 (137)
I've been unable to produce the orange transferring card at all. We may need to wait and see how a force close is handled.

**Comment by catch-21** [comment]
Managed to test it on staging. When closing a connection it provoked a force close, I needed to mine a bunch of blocks until the funds could be swept. The Transferring card showed an accurate duration throughout and got removed once the funds were swept.

Ran the test again on staging whilst closing the wallet often and letting the final blocks before sweep is possible be mined whilst the wallet is closed and everything worked as it should.

It would still be good to see this on mainnet too in a realistic timeframe, but I'm happy to close this now.
